File: cote-exercise/src/main/java/cote/baekjoon/step1/NO10828.java
package cote.baekjoon.step1;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Stack;

public class NO10828 {
    public static void main(String[] args) throws IOException {
        Stack<Integer> stack = new Stack<>();
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = br.read();

        for (int i = 0; i < n; i++) {
            String[] values = br.readLine().split(" ");
            if (values.length == 2) {
                stack.push(Integer.valueOf(values[1]));
                continue;
            }

            switch (values[0]) {
                case "pop": {
                    if (stack.isEmpty()) System.out.println(-1);
                    else System.out.println(stack.pop());
                    break;
                }
                case "size": {
                    System.out.println(stack.size());
                    break;
                }
                case "empty": {
                    if (stack.isEmpty()) System.out.println(1);
                    else System.out.println(0);
                    break;
                }
                case "top": {
                    if (stack.isEmpty()) System.out.println(-1);
                    else System.out.println(stack.peek());
                    break;
                }
            }
        }

        br.close();
    }
}

File: cote-py/baekjoon/step1/NO10828.py
import sys
from collections import deque

n = int(sys.stdin.readline())
stack = deque()

for _ in range(n):
    split = sys.stdin.readline().split()

    if split[0] == 'push':
        stack.append(split[1])
    elif split[0] == 'pop':
        if len(stack) == 0:
            print(-1)
        else:
            print(stack.pop())
    elif split[0] == 'size':
        print(len(stack))
    elif split[0] == 'empty':
        if len(stack) == 0:
            print(1)
        else:
            print(0)
    elif split[0] == 'top':
        if len(stack) == 0:
            print(-1)
        else:
            print(stack[-1])